pkg/server/server.go

```diff
@@ -132,6 +132,12 @@ func runControllers(ctx context.Context, config *Config) error {
 		}
 	}
 
+	for _, controller := range config.Controllers {
+		if err := controller(ctx, sc); err != nil {
+			return errors.Wrap(err, "controller")
```

Comment:
Does the errors returned from these controllers indicate which controller threw the error? If not, there might be some value in making the "CustomControllers" type a `map[string]func(ctx context.Context, sc *server.Context) error` with the name of the controller as the key and include the key in this error string.

Reply:
Yes, the error should be a bubbled-up/wrapped error that would indicate what controller actually threw the error when attempting to start.